### ISSUE TITLE
[BUG] Categorify does not set non-frequent item mapping size in unique parquet file when `max_size` arg is set

### ISSUE THREAD

**Issue by rnyak** [NVIDIA-Merlin/NVTabular#1735]
**Describe the bug**

When we set the `max_size` arg in Categorify() op, we intent to apply frequency capping, i.e., non-frequent items are mapped to 0, based on `max_size` value that is set. However, although Categorify works fine and encodes the frequent and non-frequent items properly, the `unique_...parquet` file does not set the size of non_frequent items correct. This is an important issue since users rely on these `unique...parquet` files and they do further processes with them. One example is to calculate the item frequencies from `unique.item_id.parquet`  file and do some logits correction in Merlin Models.

**Steps/Code to reproduce bug**
Please run the code to repro the issue

```
import nvtabular as nvt
from merlin.core import dispatch
from merlin.core.dispatch import make_df
from nvtabular import ColumnSelector, ops
import pandas as pd

df = dispatch.make_df(
    {
        "item_id": [
            "A",
            "E",
            "B",
            "C",
            "A",
            "A",
            "B",
            "C",
            "D",
            "A",
            "B",
            "A",
            "B",
        ],
    }
)

cat_names = ["item_id"]
dataset = nvt.Dataset(df)
cat_features = cat_names >> ops.Categorify(max_size=3)
processor = nvt.Workflow(cat_features)
processor.fit(dataset)
new_gdf = processor.transform(dataset).to_ddf().compute()
pd.read_parquet('./categories/unique.item_id.parquet')

	item_id	item_id_size
0	<NA>	0
1	A	5
2	B	4
```

**Expected behavior**
The size value in the first row in the `unique.item_id.parquet` should read as `4` instead of `0`, since after setting `max_size` arg, we have 4 samples in the processed `new_gdf` that were mapped to `0`.

**Environment details (please complete the following information):**
I am using `merlin-tensorflow:22.11`  image with the latest main branches of the Merlin libs pulled and installed.

**Comment by viswa-nvidia** [comment]
@rnyak , link the PR and close it

**Comment by rnyak** [comment]
closing, since this should be addressed by this PR https://github.com/NVIDIA-Merlin/NVTabular/pull/1692
